controllers/azurecluster_reconciler.go

```diff
@@ -141,6 +141,10 @@ func (s *azureClusterService) Delete(ctx context.Context) error {
 
 	if err := s.groupsSvc.Delete(ctx); err != nil {
 		if errors.Is(err, azure.ErrNotOwned) {
+			if err := s.bastionSvc.Delete(ctx); err != nil {
```

Comment:
Had to move this ahead in order, because the bastion creates dependencies on earlier services and would cause the cluster resources to not be deleted when the resource group is not managed, when resources in the cluster are deleted individually.